packages/core/react-dnd/src/decorators/decorateHandler.tsx

```diff
@@ -168,6 +168,7 @@ export default function decorateHandler<Props, CollectedProps, ItemIdType>({
 			this.disposable.dispose()
 			if (this.handlerConnector) {
 				this.handlerConnector.receiveHandlerId(null)
+				this.handlerConnector.disconnect()
```

Comment:
This is technically a `noop` because `receiveHandlerId(null)` disconnects internally, but I think it's good to leave this here as a safety measure in case the behavior changes and so that we explicitly know that we're disconnecting.